rtrlib/transport/ssh/ssh_transport.c

```diff
@@ -86,10 +89,47 @@ int tr_ssh_open(void *socket)
 		}
 	}
 
-	if (ssh_connect(ssh_socket->session) == SSH_ERROR) {
-		SSH_DBG1("tr_ssh_init: opening SSH connection failed", ssh_socket);
-		goto error;
-	}
+	ssh_set_blocking(ssh_socket->session, 0);
+	int ret;
+
+	do {
+		ret = ssh_connect(ssh_socket->session);
+
+		if (ret == SSH_ERROR) {
+			SSH_DBG1("tr_ssh_init: opening SSH connection failed", ssh_socket);
+			goto error;
+		} else if (ret == SSH_AGAIN) {
+			socket_t fd = ssh_get_fd(ssh_socket->session);
+
+			fd_set rfds;
+
+			FD_ZERO(&rfds);
+			FD_SET(fd, &rfds);
+			struct timeval tv = { 30, 0 };
```

Comment:
add comment that this is a timeout of 30s for `select` below. Also maybe change variable name to `timeout`?

Reply:
> add comment that this is a timeout of 30s for `select` below. Also maybe change variable name to `timeout`?

+1. agreed. by the way, having the timeout value configurable would be a nice to have. ( separate ticket, or separate commit )

Reply:
> add comment that this is a timeout of 30s for `select` below. Also maybe change variable name to `timeout`?

done


> > add comment that this is a timeout of 30s for `select` below. Also maybe change variable name to `timeout`?
> 
> +1. agreed. by the way, having the timeout value configurable would be a nice to have. ( separate ticket, or separate commit )

Yea, making this configurable is probably a good idea. Will add this to this PR.